website/content/docs/v0.9/Guides/converting-control-plane.md

```diff
@@ -0,0 +1,247 @@
+---
+title: "Converting Control Plane"
+description: "How to convert Talos self-hosted Kubernetes control plane (pre-0.9) to static pods based one."
+---
+
+Talos version 0.9 runs Kubernetes control plane in a new way: static pods managed by Talos.
+Talos version 0.8 and below runs self-hosted control plane.
+After Talos OS upgrade to version 0.9 Kubernetes control plane should be converted to run as static pods.
+
+This guide describes automated conversion script and also shows detailed manual conversion process.
+
+## Automated Conversion
+
+First, make sure all nodes are updated to Talos 0.9:
+
+```bash
+$ kubectl get nodes -o wide
+NAME                     STATUS   ROLES                  AGE   VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE         KERNEL-VERSION   CONTAINER-RUNTIME
+talos-default-master-1   Ready    control-plane,master   58m   v1.20.4   172.20.0.2    <none>        Talos (v0.9.0)   5.10.19-talos    containerd://1.4.4
+talos-default-master-2   Ready    control-plane,master   58m   v1.20.4   172.20.0.3    <none>        Talos (v0.9.0)   5.10.19-talos    containerd://1.4.4
+talos-default-master-3   Ready    control-plane,master   58m   v1.20.4   172.20.0.4    <none>        Talos (v0.9.0)   5.10.19-talos    containerd://1.4.4
+talos-default-worker-1   Ready    <none>                 58m   v1.20.4   172.20.0.5    <none>        Talos (v0.9.0)   5.10.19-talos    containerd://1.4.4
+```
+
+Start the conversion script:
+
+```bash
+$ talosctl -n <IP> convert-k8s
+discovered master nodes ["172.20.0.2" "172.20.0.3" "172.20.0.4"]
+current self-hosted status: true
+gathering control plane configuration
+aggregator CA key can't be recovered from bootkube-boostrapped control plane, generating new CA
+patching master node "172.20.0.2" configuration
+patching master node "172.20.0.3" configuration
+patching master node "172.20.0.4" configuration
+waiting for static pod definitions to be generated
+waiting for manifests to be generated
+Talos generated control plane static pod definitions and bootstrap manifests, please verify them with commands:
+    talosctl -n <master node IP> get StaticPods.kubernetes.talos.dev
+    talosctl -n <master node IP> get Manifests.kubernetes.talos.dev
+
+bootstrap manifests will only be applied for missing resources, existing resources will not be updated
```

Comment:
What does this mean? Missing from where?  What bootstrap manifests?  Resources existing where?  Is this something I (the user) need to worry about?

Reply:
it is explained more around lines 108-112, but probably script itself deserves better message